thorlcr/activities/hashdistrib/thhashdistribslave.cpp

```diff
@@ -2565,7 +2580,22 @@ class CSpill : public CSimpleInterface, implements IRowWriter
         OwnedIFile iFile = createIFile(tempname.str());
         spillFile.setown(new CFileOwner(iFile.getLink()));
         if (owner.getOptBool(THOROPT_COMPRESS_SPILLS, true))
+        {
             rwFlags |= rw_compress;
+            StringBuffer compType;
+            owner.getOpt(THOROPT_COMPRESS_SPILL_TYPE, compType);
+            if (compType.length())
+            {
+                if (0 == stricmp("FLZ", compType))
+                    rwFlags |= rw_fastlz;
+                else if (0 == stricmp("LZ4", compType))
+                    rwFlags |= rw_lz4;
+                else if (0 == stricmp("DEN", compType))
+                    rwFlags |= rw_den;
```

Comment:
As this mapping is / will be common, it would be good to put this code in a helper method somewhere.